rfcs/20211207-softmax-v2/README.md

```diff
@@ -0,0 +1,192 @@
+# RFC: Softmax v2
+
+## Introduction
+
+Several primitives in the library were designed with operation descriptors
+containing only `data_desc` member, which covers source and destination memory
+descriptors. There was an assumption that such operations do not change memory
+format layout through execution nor require a different output data type. And
+never would. Those primitives are batch and layer normalizations, eltwise,
+prelu, lrn, shuffle and softmax.
+
+These operations, except shuffle, assume that only floating-point types will be
+used on input and output of same data type will be produced. It is quite
+essential due to the nature and service of these operations, especially,
+normalizations. Also, outputs of these operations are clearly distinguished only
+in floating-point range of values.
+
+Unfortunately, that assumption did not consider int8 flows when computations may
+contain mixture of quantized data and floating-point data. Mentioned primitives
+are still executed in floating-point data types such as `f32`, `bf16` or `f16`,
```

Comment:
Is the plan to support all combinations f32/bf16/f16 -> u8/s8?

Reply:
Yes. And even more - all data types to all data types. Might be bit more complicated for int8 output, but no big reasons not to support any possible combination since kernel is the same.

Reply:
For u8/s8 inputs, do we need to support output scale?

Reply:
Seems like I was a bit ambitious in the plan. I'll focus on fp -> int8 things first, and restrict int8 -> any cases for now until they are needed and used. Then we think about it better. Thanks for the point raised.

Reply:
Yeah, sounds good to me, thanks!